fe/fe-core/src/main/java/com/starrocks/catalog/Function.java

```diff
@@ -442,6 +442,10 @@ private boolean isIndistinguishable(Function o) {
             if (o.argTypes[i].isNull() || this.argTypes[i].isNull()) {
                 continue;
             }
+            if (FunctionSet.TRUNCATE.equalsIgnoreCase(name.getFunction()) &&
```

Comment:
This is seems very strange.

Reply:
Yes, otherwise it will always match double version, do you have any suggestion?